content/posts/2024-09-01-reviving-newgvn.md

```diff
@@ -0,0 +1,33 @@
+---
+author: "Manuel Brito"
+date: "2024-09-01"
+tags: ["GSoC", "optimization"]
+title: "GSoC 2024: Reviving NewGVN"
+---
+
+This summer I participated in GSoC under the LLVM Compiler Infrastructure. The goal of the project was to improve the NewGVN pass so that it can replace GVN as the main value numbering pass in LLVM.
+
+# Motivation
+
+NewGVN was introduced in 2016 to replace GVN. NewGVN improves upon GVN by not relying on MemoryDependenceAnalysis to detect redundancies among memory operations, using a sparse formulation, being complete for loops, among other aspects. However, it is still not widely used, mainly because it lacks partial redundancy elimination (PRE) and is bug-ridden.
```

Comment:
could mention MemorySSA & optimistic vs pessimistic approach. Could give an example.

Reply:
I added a background section explaining how MemorySSA improves NewGVN and also clarified the difference between optimistic and pessimistic value numbering.